apps/web-portal/lib/WebPortalExpressApp.mjs

```diff
@@ -0,0 +1,39 @@
+import assert from 'assert'
+import express from 'express'
+import {appEndpoint} from './appEndpoint.mjs'
+import {makePreview} from './preview/makePreview.mjs'
+
+// TODO: gh-55 handle process start/stop, wire with logger, etc
+export class WebPortalExpressApp {
+
+  /**
+   * @type {Express}
+   */
+
+  #http = null
+  #config = null
+
+  /**
+   * @param {Object} context
+   * @param {Object} context.config - app config
+   */
+
+  constructor(context = {}) {
+    assert(context.config, 'Should have an app config defined')
+
+    this.#config = context.config
+    this.#http = express()
+
+    this.initEndpoints()
+  }
+
+  initEndpoints() {
+    appEndpoint(this.#http, '/previews', makePreview)
```

Comment:
maybe move this endpoints adding to another file? endpoints configuration could be very specific.

Reply:
Yep, will do, not sure yet about endpoint, thinking about just a factory function with direct express router